Automattic/woocommerce-payments
# Fix the checkout in IE11

Fixes #821

#### Changes proposed in this Pull Request

In this PR, I added the `wcpay-checkout.js` script to our Webpack + Babel build system. That means that JS syntax unsupported in IE11 will be transpiled and just work. Going commit-by-commit:
- https://github.com/Automattic/woocommerce-payments/commit/1bd9281eecc283f6f13912d644e23cb6a58f88e1 Move the script and CSS files to the build system.
- https://github.com/Automattic/woocommerce-payments/commit/2dec2d19aefbe3dcfc76bbb99c36aadd4581e4de Removed a property access that's not supported on IE11. Babel can't protect us from everything.
- https://github.com/Automattic/woocommerce-payments/commit/ee2f7bbfda911bf3825ec0d719db790626a138c8 https://github.com/Automattic/woocommerce-payments/commit/78acf1af10f70cae50768749fcd89a28f1a432fa made the checkout script a bit more compliant with our "modern" code style.

I realized how stuborn I've been about not wanting to add the checkout to the Webpack+Babel pipeline. My reasoning was _"on the frontend, every KB counts, specially in something as mission-critical as the checkout. Webpack+Babel will add too much bloat"_. Which made sense, but here's the kicker: **We aren't running the minifier on `/assets/js/`**. So, the hand-crafted `wcpay-checkout.js` file was around `15KB`. Now that it goes through the whole Webpack+Babel contraption, it's **6.5KB**. Of course, applying the minifier **and** not running Webpack+Babel would give us something even smaller than that, but at that point it's just diminishing returns.

So, by adding the checkout to the build pipeline, it's actually getting smaller. And we won't need to worry about using JS syntax that's "too modern" for IE11. Win-win.

#### Testing instructions

I tested the checkout on a [Windows 7 VM with IE11 installed](https://developer.microsoft.com/en-us/microsoft-edge/tools/vms/), but you don't have to. You could just check that nothing broke in Chrome or whatever modern browser you use. Test cases:
- Pay with an SCA card.
- Pay with a regular card.
- Pay with a saved card.

Props @luizreis for doing the legwork on this issue!
